dbms/src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -538,7 +538,7 @@ BlockIO InterpreterCreateQuery::createTable(ASTCreateQuery & create)
             false);
 
         if (create.is_temporary)
-            context.getSessionContext().addExternalTable(table_name, res);
+            context.getSessionContext().addExternalTable(table_name, res, query_ptr);
```

Comment:
After `CREATE ... AS SELECT` queries you will have `SELECT` part present in the result of `SHOW CREATE` query. This goes in contrast with results of `SHOW CREATE` queries for ordinary tables: databases don't store the `SELECT` part and don't show it in `SHOW CREATE`.